compiler/rustc_trait_selection/src/solve/delegate.rs

```diff
@@ -136,6 +136,15 @@ impl<'tcx> rustc_next_trait_solver::delegate::SolverDelegate for SolverDelegate<
                     None
                 }
             }
+            ty::PredicateKind::Clause(ty::ClauseKind::WellFormed(arg)) => {
+                if arg.is_trivially_wf(self.tcx) {
+                    Some(Certainty::Yes)
+                } else if self.shallow_resolve_term(arg).is_infer() {
```

Comment:
did u try flipping the order in this check? i.e. first shallow resolve, then check for trivial wf

Reply:
oh duh lol

no, i only added the shallow resolve after but it definitely could benefit or at least shouldn't hurt